### DISCUSSION TITLE
Using multiple chat services in one Kernel for different skills

### DISCUSSION THREAD

**Discussion by VladTolm** [microsoft/semantic-kernel#1926] [Q&A]
Maybe I'm doing something wrong, but it doesn't seem to work in the Python version reading the "default_backends" key for skills.

No matter what values I put there, the default chat service is still used.

In the tests and documentation I did not find examples of using different services for different skills

**Comment by aeharriman** [comment]
I am also encountering this issue. Using the kernel.set_default_chat_service() before calling semantic functions does not work either.

**Comment by aeharriman** [comment]
If anyone else ever runs into this, setting the default chat service actually ended up working. In my previous comment I was:

1. Using `kernel.set_default_chat_service("service_id)` when I should have been using `kernel.set_default_text_completion_service("service_id")`
2. Using it right before using `kernel.run_async()` with my semantic function I was trying to specify the `service_id` for.

What ended up working was using `kernel.set_default_text_completion_service("service_id")` before importing the respective skills into the kernel, like this:

    kernel.set_default_text_completion_service("service_1")
    kernel.import_semantic_skill_from_directory(skills_directory, "Plugin I wanted to use with service 1")

    kernel.set_default_text_completion_service("service_2")
    kernel.import_semantic_skill_from_directory(skills_directory, "Plugin I wanted to use with service 2")

**Comment by punitarani** [comment]
I still think it would be nice to have a param or config option for each skill to indicate the service preference
